afs/views/workflows/add_chemical_analysis_images/add_chemical_analysis_images_file_upload.py

```diff
@@ -0,0 +1,175 @@
+from io import FileIO
+from mimetypes import MimeTypes
+from django.views.generic import View
+from django.db import transaction
+import os
+import io
+from arches.app.datatypes.datatypes import StringDataType
+from arches.app.utils.response import JSONResponse
+from arches.app.views.tile import TileData
+from arches.app.models.tile import Tile
+from arches.app.models.resource import Resource
+from arches.app.models.models import Node
+from arches.app.utils.betterJSONSerializer import JSONSerializer, JSONDeserializer
+import json
+from django.core.files.uploadedfile import InMemoryUploadedFile
+from django.http import HttpRequest
+from django.conf import settings
+
+
+class AddChemicalAnalysisImagesFileUpload(View):
+    def post(self, request):
+        recorded_value_node_id = "dd596aae-c457-11e9-956b-a4d18cec433a"
+        observation_digital_source_nodegroup_id = "0ae149ba-8e30-11eb-a9c4-faffc265b501"
+        observation_digital_source_node_id = "0ae14d2a-8e30-11eb-a9c4-faffc265b501"
+        digital_reference_node_id = "a298ee52-8d59-11eb-a9c4-faffc265b501"
+        digital_reference_node_group_id = "8a4ad932-8d59-11eb-a9c4-faffc265b501"
+        dataset_name_node_group_id = "d2fdae3d-ca7a-11e9-ad84-a4d18cec433a"
+        dataset_name_node_id = "d2fdc2fa-ca7a-11e9-8ffb-a4d18cec433a"
+        dataset_file_node_group_id = "7c486328-d380-11e9-b88e-a4d18cec433a"
+        dataset_file_node_id = "7c486328-d380-11e9-b88e-a4d18cec433a"
+
+        transaction_id = request.POST.get("transaction_id")
+        observation_id = request.POST.get("observation_id")
+        digital_resource_instance_ids = []
+        # posted_dataset = request.POST.get("dataset")
+        # observation_ref_tile = request.POST.get("observation_ref_tile", None)
+
+        files = request.FILES.getlist("file-list_{}_preloaded".format(dataset_file_node_id), [])
+        removed_files = []
+        # Create dataset for each file
+        # 'Upload' files
+        # Relate each dataset to existing observation
+
+
+        with transaction.atomic():
```

Comment:
Functions that need the transaction atomic block need to be called within the block rather than defined within it. These functions need to defined above the block and where they are called down on line 160, needs to be moved within the block.